src/components/include/utils/custom_string.h

```diff
@@ -180,6 +180,20 @@ class CustomString {
   bool CompareIgnoreCase(const char* str) const;
 
   /**
+   * @brief Compares the value of the string (case sensitive).
```

Comment:
@dev-gh Right but these changes are critical and needed for further development.